synapse/models/inet.py

```diff
@@ -1442,7 +1442,11 @@ def getModelDefs(self):
                         'doc': 'A channel within a web service or instance such as slack or discord.'
                     }),
 
-                    ('inet:web:hashtag', ('str', {'lower': True, 'regex': r'^#\w[\w·]*(?<!·)$'}), {
+                    ('inet:web:hashtag', ('str', {'lower': True, 'regex': r'^#[^\p{Z}#]+$'}), {
```

Comment:
```suggestion
                    ('inet:web:hashtag', ('str', {'lower': True, 'strip': True, 'regex': r'^#[^\p{Z}#]+$'}), {
```

Reply:
Adding strip would require us to modify the tests were whitespace is appended to end.